### DISCUSSION TITLE
[译] ProseMirror 中文指南

### DISCUSSION THREAD

**Discussion by Xheldon** [Xheldon/x_blog#125] [Announcements]
https://www.xheldon.com/prosemirror-guide-chinese.html 

 [译] ProseMirror 中文指南

**Comment by Xheldon** [comment]
欢迎留言!

**Comment by wangwenbo000** [comment]
感谢你的翻译！辛苦了

**Comment by SolidZORO** [comment]
老大，你这 body 的 font-weight 由 300 改成 400，看起来会舒服很多，没那么累眼了。现在我每次看都手动调一下。 😭

**Comment by Xheldon** [comment]
> 老大，你这 body 的 font-weight 由 300 改成 400，看起来会舒服很多，没那么累眼了。现在我每次看都手动调一下。 😭

哈哈哈，我是参照苹果官网的 font-weight 设置的，不过...苹果的也是 400，那我改下好了😆

**Comment by Xheldon** [comment]
关了 issue gtalk 就无法显示在网页留言了。。

**Comment by SolidZORO** [comment]
还有这种事情，赶紧给你的 comment 服务商 issue 一波。

**Comment by Brass-neck** [comment]
老大，关于collaborative这块怎么和ot算法结合呢？可否给小弟指导二三，万分感谢

**Comment by Xheldon** [comment]
> 老大，关于collaborative这块怎么和ot算法结合呢？可否给小弟指导二三，万分感谢

这块我也接触的不深，目前知道的是服务端用一个 ot 算法解决冲突（可能是服务端的 yjs？），然后 pm 中写一个插件用 yjs 实现通讯。
作者在  https://github.com/ProseMirror/website/issues/63 说过要写一个协同代码说明，但是至今仍未写...好几年了，你可以给他发邮件催一催，哈哈

**Comment by hidumou** [comment]
> 你会经常想要和 node 交互以影响 document 中的真实 node. 但是为了创建一个 transaction 去改变一个 node, 你首先需要知道这个 node 在哪儿. 为了让你能够做到这一点, node views 传递了一个`getter` 函数 可以用来查询它们当前在 document 中的位置. 让我们修改下刚刚的示例, 使它能够在点击这个 node 的时候让你为这个 image 节点输入 alt 信息.

貌似写错了，不是传递了一个 `getPos` 么

**Comment by Xheldon** [comment]
@haner199401  [原文](https://prosemirror.net/docs/guide/#:~:text=You%27ll%20often%20want%20interaction,position%20in%20the%20document.)是这样的，我就直接翻译过来了：
> You'll often want interaction with the node to have some effect on the actual node in the document. But to create a transaction that changes a node, you first need to know where that node is. To help with that, node views get passed a getter function that can be used to query their current position in the document.

**Comment by tolerious** [comment]
支持点赞。

**Comment by tolerious** [comment]
我写了一个简单的 Demo 包含一些基本的示例方便做实验, 放在了: 这个仓库, 欢迎fork/star.      仓库地址好像失效了。

**Comment by Xheldon** [comment]
> @tolerious
> 我写了一个简单的 Demo 包含一些基本的示例方便做实验, 放在了: 这个仓库, 欢迎fork/star.      仓库地址好像失效了。

Sorry, 这个仓库我删除了，因为这个仓库就是官方的那个示例仓库，加了点 Webpack 的配置以及打印了 log ，没有其他的修改，你直接用官方的那个示例就可以。

**Comment by kingshuaishuai** [comment]
作者给的Decorations的第二个例子是跑不通的，直接在 decorations 中写那段逻辑可以，通过 state 获取就不行了，我还以为是版本问题，结果去英文官网一看也是相同的例子，跑不通，但不太明白原因

**Reply by tolerious** [reply]
@kingshuaishuai   给个链接

**Comment by alwmiku** [comment]
在Serialization and Parsing这一节中有一段话:

上面示例中, [“p”, 0] 的含义是 paragraph 节点在 HTML 中被渲染成 <p> 标签. 0 代表一个 “hole”, 表示该 node 的内容应该被渲染的地方(意思就是如果这个节点预期是有内容的, 就应该在数组最后写上 0);

这个译者的注释的括号里的话有问题理解不对。这里的0和数组没关系；而是指的渲染位置。原意的翻译没错，加了注释反而不懂了，
我觉得应该翻译成：
（意思就是如果这个节点预期是有内容的, 那会在“0”位置渲染对应内容）

这个0位置具体是什么我也说不太清，这句的意思就是说有内容的话，内容就被会一个<p>标签包裹住。

**Reply by Xheldon** [reply]
没错，是你这么理解的
